### DISCUSSION TITLE
[WEB-69556] Better support for Vue template

### DISCUSSION THREAD

**Discussion by ZYong9908** [izhangzhihao/intellij-rainbow-brackets#2704] [Q&A] [answered]
![image](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/44081531/9057eb22-ebcd-4bac-a8b9-3ef0a8fd2f3b)
![image](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/44081531/48689078-f27b-4361-82ad-a396432c1884)
![image](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/44081531/f1647401-86cc-4f28-b76b-b0d18181511d)
在webstorme中编辑.vue文件，原生HTML标签颜色改变正常，自定义组件/第三方ui组件只会改变尖括号颜色，不会改变标签名称颜色

**Comment by izhangzhihao** [comment]
pls provide a sample code so I can re-produce it.

**Reply by ZYong9908** [reply]
you can use the example to  re-produce in any VUE project:
```
<script>
import test_sec from './test_sec.vue'
export default {
  name: 'test_main',
  components: { test_sec }
}
</script>

<template>
<div>
  <test_sec></test_sec>
</div>
</template>
```
```
<script>
export default {
  name: 'test_sec'
}
</script>

<template>
<div>test</div>
</template>
```
![image](https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/44081531/e4d19f4a-7837-40f7-9af6-bd232315e9c1)

**Comment by izhangzhihao** [comment]
Related to https://github.com/izhangzhihao/intellij-rainbow-brackets/discussions/2645 ,  I can reproduce it in my IDE now, because of the `template` tag and others cannot be recognized as HTML/XML syntax, I will try to support this later.

**Comment by ZYong9908** [comment]
thanks, looking forward to updates

Zhihao Zhang ***@***.***> 于 2024年1月19日周五 18:36写道：

**Comment by izhangzhihao** [comment]
I did some testing here:

1. sometimes it seems to work fine but sometimes does not :
<img width="629" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/7754a2cf-90f3-4908-beb4-46fafd06bd38">

2. change the file type or file extension to XML the highlight works as expected: (so there is nothing wrong with the HTML/XML highlighter)
<img width="836" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/64d23b91-1c79-4def-94c5-2cc6fcd60561">
<img width="669" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/f6421128-5d6f-4ca2-8546-07ef5dc44a0b">


3. so I downloaded the Vue plugin, and unzipped it, I found [VueFileHighlighter](https://github.com/JetBrains/intellij-plugins/blob/master/vuejs/src/org/jetbrains/vuejs/lang/html/highlighting/VueFileHighlighter.kt) inside the plugin, it works like the HTML highlighter but did some extra highlighting for the Vue files. I guess when doing Vue highlighting, this plugin may conflict with the Vue plugin. So it may looks like sometimes works sometimes does not.

4. I didn't find any way to config the Vue plugin to stop highlighting Vue template keywords but I will keep an eye on this.

**Reply by ZYong9908** [reply]
this is similar to my guess, if i find anything useful i will share it here

**Comment by izhangzhihao** [comment, marked as answer]
Revisited and checked, it works as expected in 2025.3 EAP

<img width="911" height="944" alt="image" src="https://github.com/user-attachments/assets/3d0f227c-57e7-46df-9efa-57da29318e83" />
